### PR TITLE
feat: Detect Safari web view alerts

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Related to https://github.com/appium/appium-xcuitest-driver/pull/1185

### CODE REVIEW

**Comment by imurchie** [comment]
This works for alerts (with a single button) but not prompts. Didn't test against the allow/block style popup. Things are pretty slow, too, though this is pretty true throughout the alert handling in Safari, unfortunately.

**Comment by mykola-mokhnach** [comment]
@imurchie Cool, I've added more logging. Please attach the log with xcode logging enabled for cases where the alert is not found

**Comment by mykola-mokhnach** [comment]
Now the performance should be slightly better. I've also tested the recent implementation on https://www.seleniumeasy.com/test/javascript-alert-box-demo.html and all alerts worked

**Comment by imurchie** [comment]
Failures:
1. Running https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/web/safari-alerts-e2e-specs.js#L59-L66
    logs: https://gist.github.com/imurchie/fe2249fbfb575cbc8f17c33e65b88cc6

Issues:
1. Slowness in https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/web/safari-window-e2e-specs.js#L74-L80
    logs: https://gist.github.com/imurchie/7f0b67facb1c1eac6c912df7d344cf5e


Also, the error handling for prompts currently returns 
```
info [09-54-00:933] WD Proxy Got response with status 500: {"value":{"error":"unknown error","message":"Error Domain=com.facebook.WebDriverAgent Code=1 \"The alert contains no input fields\" UserInfo={NSLocalizedDescription=The alert contains no input fields}","traceback":"(\n\t0   WebDriverAgentLib                   0x0000000129a89ba1 +[FBAlertViewCommands handleAlertSetTextCommand:] + 769\n\t1   WebDriverAgentLib                   0x0000000129a638c6 -[FBRoute_TargetAction mountRequest:intoResponse:] + 182\n\t2   WebDriverAgentLib                   0x0000000129a4eadb __37-[FBWebServer registerRouteHandlers:]_block_invoke + 555\n\t3   RoutingHTTPServer                   0x0000000129b55d9f -[RoutingHTTPServer handleRoute:withRequest:response:] + 136\n\t4   RoutingHTTPServer                   0x0000000129b566b2 __72-[RoutingHTTPServer routeMethod:withPath:parameters:request:connection:]_block_invoke + 47\n\t5   libdispatch.dylib                   0x00000001102e57b9 _dispatch_client_callout + 8\n\t6   libdispatch.dylib                   0x00000001102f2861 _dispatch_a...
```
but should follow https://www.w3.org/TR/webdriver/#send-alert-text

Note, for reading the logs, I added a call to get the alert text in order to check for the presence of an alert during atom execution.

**Comment by mykola-mokhnach** [comment]
I think the slowness might be observed because of the existing alert detection logic. I did a couple of optimizations for that in https://github.com/appium/WebDriverAgent/pull/298.

Let me merge that PR here, so we could check the resulting performance

**Comment by imurchie** [comment]
Ok! Merge and rebase and we can see.

**Comment by mykola-mokhnach** [comment]
the issue with text input has been addressed. Also the overall speed is also better now as well

**Comment by imurchie** [comment]
This works ok (i.e., about the same performance as the old way) on iOS 13.3. Fails for 12.2 and 12.1, though the existing logic works for all of them.

Test run: https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/web/safari-alerts-e2e-specs.js#L36-L40
Logs: https://gist.github.com/imurchie/181c8d35d3184e8fe649274d053c8fdd

You can also see this in action (without the Xcode logs) in a travis build I kicked off linked to this WDA branch: https://travis-ci.org/github/imurchie/appium-xcuitest-driver/builds/666822720

**Comment by imurchie** [comment]
The source reported by `GET http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/source`
```xml
<?xml version="1.0" encoding="UTF-8"?>
<AppiumAUT>
<XCUIElementTypeApplication type="XCUIElementTypeApplication" name="Safari" label="Safari" enabled="true" visible="true" x="0" y="0" width="375" height="812">
  <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="false" x="0" y="0" width="375" height="812">
    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="812"/>
  </XCUIElementTypeWindow>
  <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="true" x="0" y="0" width="375" height="812">
    <XCUIElementTypeStatusBar type="XCUIElementTypeStatusBar" enabled="true" visible="true" x="0" y="0" width="375" height="44">
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="44">
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="44">
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="6" y="-5" width="180" height="12"/>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="190" y="-5" width="180" height="12"/>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="6" y="17" width="180" height="12"/>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="13" y="14" width="67" height="15">
            <XCUIElementTypeStaticText type="XCUIElementTypeStaticText" value="3:36 PM" name="3:36 PM" label="3:36 PM" enabled="true" visible="false" x="30" y="14" width="34" height="18"/>
          </XCUIElementTypeOther>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="21" y="8" width="56" height="22"/>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="190" y="17" width="180" height="12"/>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="294" y="14" width="67" height="15">
            <XCUIElementTypeOther type="XCUIElementTypeOther" value="No signal" name="Cellular" label="Cellular" enabled="true" visible="false" x="294" y="17" width="17" height="12"/>
            <XCUIElementTypeOther type="XCUIElementTypeOther" value="SSID, 3 of 3 Wi-Fi bars" name="3 of 3 Wi-Fi bars" enabled="true" visible="false" x="316" y="17" width="16" height="12"/>
            <XCUIElementTypeOther type="XCUIElementTypeOther" value="Not charging" name="100% battery power" label="100% battery power" enabled="true" visible="false" x="336" y="17" width="25" height="12"/>
          </XCUIElementTypeOther>
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="294" y="14" width="67" height="15"/>
        </XCUIElementTypeOther>
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="8" y="24" width="82" height="14"/>
      </XCUIElementTypeOther>
    </XCUIElementTypeStatusBar>
  </XCUIElementTypeWindow>
</XCUIElementTypeApplication>
</AppiumAUT>
```
However, the required elements are there. Here is the logging from accepting the alert during which the source was retrieved:
```
info [15-36-01:608] HTTP --> POST /wd/hub/session/8b29c987-b80f-48fd-a62d-aa013313be38/accept_alert
info [15-36-01:608] HTTP {}
dbug [15-36-01:609] MJSONWP (8b29c987) Calling XCUITestDriver.postAcceptAlert() with args: ["8b29c987-b80f-48fd-a62d-aa013313be38"]
dbug [15-36-01:609] XCUITest Executing command 'postAcceptAlert'
dbug [15-36-32:488] WD Proxy Matched '/alert/accept' to command name 'postAcceptAlert'
dbug [15-36-32:488] WD Proxy Proxying [POST /alert/accept] to [POST http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/alert/accept] with body: {}
info [15-37-02:850] WD Proxy Got response with status 404: {"value":{"error":"no such alert","message":"An attempt was made to operate on a modal dialog when one was not open","traceback":""},"sessionId":"A99E1D1F-F1A5-4F34-8CB3-CF6992114454"}
dbug [15-37-02:851] W3C Matched W3C error code 'no such alert' to NoSuchAlertError
dbug [15-37-02:851] BaseDriver Waiting up to 0 ms for condition
dbug [15-37-02:852] WD Proxy Matched '/elements' to command name 'findElements'
dbug [15-37-02:852] WD Proxy Proxying [POST /elements] to [POST http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/elements] with body: {"using":"class name","value":"XCUIElementTypeScrollView"}
dbug [15-37-12:915] WD Proxy Got response with status 200: {"value":[{"ELEMENT":"25000000-0000-0000-8F7C-010000000000","element-6066-11e4-a52e-4f735466cecf":"25000000-0000-0000-8F7C-010000000000"}],"sessionId":"A99E1D1F-F1A5-4F34-8CB3-CF6992114454"}
dbug [15-37-12:915] BaseDriver Waiting up to 0 ms for condition
dbug [15-37-12:915] WD Proxy Matched '/element/25000000-0000-0000-8F7C-010000000000/elements' to command name 'findElementsFromElement'
dbug [15-37-12:915] WD Proxy Proxying [POST /element/25000000-0000-0000-8F7C-010000000000/elements] to [POST http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/element/25000000-0000-0000-8F7C-010000000000/elements] with body: {"using":"class name","value":"XCUIElementTypeButton"}
dbug [15-37-20:278] XCUITest Error received while executing atom: operation timed out
dbug [15-37-33:040] WD Proxy Got response with status 200: {"value":[{"ELEMENT":"3E000000-0000-0000-8F7C-010000000000","element-6066-11e4-a52e-4f735466cecf":"3E000000-0000-0000-8F7C-010000000000"}],"sessionId":"A99E1D1F-F1A5-4F34-8CB3-CF6992114454"}
dbug [15-37-33:041] WD Proxy Matched '/element/3E000000-0000-0000-8F7C-010000000000/attribute/name' to command name 'getAttribute'
dbug [15-37-33:041] WD Proxy Proxying [GET /element/3E000000-0000-0000-8F7C-010000000000/attribute/name] to [GET http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/element/3E000000-0000-0000-8F7C-010000000000/attribute/name] with no body
dbug [15-37-43:109] WD Proxy Got response with status 200: {
dbug [15-37-43:109] WD Proxy   "value" : "Close",
dbug [15-37-43:109] WD Proxy   "sessionId" : "A99E1D1F-F1A5-4F34-8CB3-CF6992114454"
dbug [15-37-43:109] WD Proxy }
dbug [15-37-43:111] WD Proxy Matched '/element/3E000000-0000-0000-8F7C-010000000000/click' to command name 'click'
dbug [15-37-43:111] WD Proxy Proxying [POST /element/3E000000-0000-0000-8F7C-010000000000/click] to [POST http://127.0.0.1:8100/session/A99E1D1F-F1A5-4F34-8CB3-CF6992114454/element/3E000000-0000-0000-8F7C-010000000000/click] with no body
dbug [15-38-03:516] RemoteDebugger Handling message (id: '16')
dbug [15-38-03:516] RemoteDebugger Received data response from send (id: 16): '"{\"status\":0,\"value\":null}"'
dbug [15-38-03:516] RemoteDebugger Sending to Web Inspector took 163227ms
dbug [15-38-03:517] RemoteDebugger Received result for atom 'click' execution: null
dbug [15-38-03:561] WD Proxy Got response with status 200: {
dbug [15-38-03:561] WD Proxy   "value" : null,
dbug [15-38-03:561] WD Proxy   "sessionId" : "A99E1D1F-F1A5-4F34-8CB3-CF6992114454"
dbug [15-38-03:561] WD Proxy }
dbug [15-38-03:562] MJSONWP (8b29c987) Responding to client with driver.postAcceptAlert() result: null
info [15-38-03:568] HTTP <-- POST /wd/hub/session/8b29c987-b80f-48fd-a62d-aa013313be38/accept_alert 200 121954 ms - 76
```

**Comment by mykola-mokhnach** [comment]
@imurchie I've downloaded iOS 12.2 Simulator and the page source, actually,  looks differently there:

```
Application, 0x600001d9b480, pid: 13398, label: 'Safari'
    Window (Main), 0x600001d9b560, {{0.0, 0.0}, {375.0, 667.0}}
      Other, 0x600001d9b640, {{0.0, 0.0}, {375.0, 667.0}}
        Other, 0x600001d9b720, {{0.0, 0.0}, {375.0, 667.0}}
          Other, 0x600001d9b800, {{0.0, 0.0}, {375.0, 667.0}}, identifier: 'BrowserWindow'
            Other, 0x600001d9b8e0, {{0.0, 20.0}, {375.0, 647.0}}
              Other, 0x600001d9b9c0, {{0.0, 20.0}, {375.0, 647.0}}
                Other, 0x600001d9baa0, {{0.0, 20.0}, {375.0, 50.0}}
                  Other, 0x600001d9bb80, {{0.0, 0.0}, {375.0, 70.0}}
                    Other, 0x600001d9bc60, {{0.0, 0.0}, {375.0, 70.0}}
                    Other, 0x600001d9bd40, {{0.0, 0.0}, {375.0, 70.0}}
                  Button, 0x600001d9be20, {{0.0, 0.0}, {375.0, 70.5}}, Disabled
                  Other, 0x600001d9bf00, {{0.0, 20.0}, {375.0, 50.0}}
                    Button, 0x600001d8a4c0, {{19.5, 24.0}, {336.0, 36.0}}, identifier: 'URL', label: 'Address'
                      Other, 0x600001d8a760, {{117.5, 32.0}, {140.0, 20.5}}, identifier: 'URL', label: 'Address'
                      Button, 0x600001d8aae0, {{329.5, 24.0}, {36.0, 36.0}}, identifier: 'ReloadButton', label: 'reload'
                  Other, 0x600001d8aa00, {{0.0, 70.0}, {375.0, 0.5}}
                ScrollView, 0x600001d8a680, {{0.0, 0.0}, {375.0, 667.0}}
                  Other, 0x600001d8a920, {{0.0, 0.0}, {375.0, 667.0}}
                    Other, 0x600001d8aca0, {{0.0, 0.0}, {375.0, 667.0}}, identifier: 'WebView'
                      Other, 0x600001d8abc0, {{0.0, 0.0}, {375.0, 667.0}}
                        Other, 0x600001d8ad80, {{0.0, 0.0}, {375.0, 667.0}}
                        Other, 0x600001d8ae60, {{20.0, 275.5}, {335.0, 142.0}}
                          Other, 0x600001d8af40, {{20.0, 275.5}, {335.0, 56.0}}
                            TextView, 0x600001d8b2c0, {{20.0, 275.5}, {335.0, 56.0}}
                          TextField, 0x600001d8b3a0, {{40.0, 329.5}, {295.0, 35.0}}
                          Other, 0x600001d8bf00, {{187.0, 364.5}, {168.0, 53.0}}
                            Button, 0x600001d8bd40, {{207.0, 374.5}, {73.0, 33.0}}, label: 'Cancel'
                            Button, 0x600001d8be20, {{290.0, 374.5}, {45.0, 33.0}}, label: 'OK'
                      WebView, 0x600001d9d960, {{0.0, 0.0}, {375.0, 667.0}}
                        Other, 0x600001d84d20, {{0.0, 0.0}, {375.0, 667.0}}
              Toolbar, 0x600001d92060, {{0.0, 623.0}, {375.0, 44.0}}, identifier: 'BottomBrowserToolbar'
                Other, 0x600001d91dc0, {{0.0, 622.5}, {375.0, 0.5}}
                Other, 0x600001d91ea0, {{0.0, 623.0}, {375.0, 44.0}}
                  Other, 0x600001d9da40, {{0.0, 623.0}, {375.0, 44.0}}
                    Button, 0x600001d84ee0, {{0.0, 623.0}, {31.0, 44.0}}, label: 'Back', Disabled
                    Button, 0x600001d85c00, {{96.0, 623.0}, {20.0, 44.0}}, label: 'Forward', Disabled
                    Button, 0x600001d857a0, {{172.0, 623.0}, {29.0, 44.0}}, label: 'Share'
                    Button, 0x600001d86220, {{249.5, 623.0}, {35.0, 44.0}}, label: 'Bookmarks'
                    Button, 0x600001d84e00, {{325.0, 623.0}, {50.0, 44.0}}, label: 'Tabs'
        Other, 0x600001d84a80, {{0.0, 0.0}, {44.0, 667.0}}
        Other, 0x600001d84700, {{331.0, 0.0}, {44.0, 667.0}}
        Other, 0x600001d85b20, {{374.5, 0.0}, {1.0, 667.0}}, identifier: 'BrowserWindowSeparator'
          Other, 0x600001d85960, {{374.5, 0.0}, {1.0, 667.0}}
    Window, 0x600001d85a40, {{0.0, 0.0}, {375.0, 667.0}}
      Other, 0x600001d86140, {{0.0, 0.0}, {375.0, 667.0}}
        Other, 0x600001d86060, {{0.0, 667.0}, {375.0, 216.0}}
    Window, 0x600001d85f80, {{0.0, 0.0}, {375.0, 667.0}}
      StatusBar, 0x600001d85ea0, {{0.0, 0.0}, {375.0, 20.0}}
        Other, 0x600001d85dc0, {{0.0, 0.0}, {375.0, 20.0}}
        Other, 0x600001d85ce0, {{0.0, 0.0}, {375.0, 20.0}}
          Other, 0x600001d848c0, {{6.0, 0.0}, {39.0, 20.0}}
          Other, 0x600001d84620, {{50.0, 0.0}, {15.0, 20.0}}, label: '3 of 3 Wi-Fi bars'
          Other, 0x600001d9db20, {{164.0, 0.0}, {50.0, 20.0}}, label: '9:42 PM'
          Other, 0x600001d9dc00, {{335.0, 0.0}, {35.0, 20.0}}, label: '86% battery power'
    Window, 0x600001d9dce0, {{0.0, 0.0}, {375.0, 667.0}}
      Other, 0x600001d9ddc0, {{0.0, 0.0}, {375.0, 667.0}}
Path to element:
 →Application, 0x600001d9b480, pid: 13398, label: 'Safari'
Query chain:
 →Find: Application 'com.apple.mobilesafari'
  Output: {
    Application, pid: 13398, label: 'Safari'
```

The alert locator has been tuned to work for both iOS versions

File: WebDriverAgentLib/Categories/XCUIApplication+FBAlert.m [hunk 1]
```diff
@@ -11,28 +11,76 @@
...
 #import "FBXCodeCompatibility.h"
 
+#define MAX_CENTER_DELTA 10.0
+
+NSString *const FB_SAFARI_APP_NAME = @"Safari";
+
+
 @implementation XCUIApplication (FBAlert)
 
+- (nullable XCUIElement *)fb_alertElementFromSafariWithScrollView:(XCUIElement *)scrollView
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Categories/XCUIApplication+FBAlert.m:22]: wow, tricky...

**Review by KazuCocoa — APPROVED** [review]
Awesome...
thanks 🎉

**Review by imurchie — APPROVED** [review]
Finally finished testing with the `appium-xcuitest-driver` Safari suite. Using Xcode 11.4, this patch works for simulators of iOS 11.0-13.4. Everything works!

**Comment by imurchie** [comment]
iOS 10.3.1 still has issues, though.

Source at time of trying to get alert text:
```xml
<?xml version="1.0" encoding="UTF-8"?><AppiumAUT><XCUIElementTypeApplication type="XCUIElementTypeApplication" name="Safari" label="Safari" enabled="true" visible="true" x="0" y="0" width="375" height="667">
  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="667"/>
  <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="true" x="0" y="0" width="375" height="667">
    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="20" width="375" height="647">
            <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="20" width="375" height="647">
              <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="20" width="375" height="44">
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="64">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="64"/>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="64"/>
                </XCUIElementTypeOther>
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="20" width="375" height="44">
                  <XCUIElementTypeButton type="XCUIElementTypeButton" value="‎localhost" name="URL" label="Address" enabled="true" visible="true" x="9" y="24" width="357" height="30">
                    <XCUIElementTypeOther type="XCUIElementTypeOther" value="‎localhost" name="URL" label="Address" enabled="true" visible="true" x="11" y="24" width="353" height="30"/>
                    <XCUIElementTypeButton type="XCUIElementTypeButton" name="ReloadButton" label="reload" enabled="true" visible="true" x="336" y="24" width="30" height="30"/>
                  </XCUIElementTypeButton>
                  <XCUIElementTypeButton type="XCUIElementTypeButton" name="Cancel" label="Cancel" enabled="true" visible="false" x="379" y="24" width="57" height="30">
                    <XCUIElementTypeStaticText type="XCUIElementTypeStaticText" value="Cancel" name="Cancel" label="Cancel" enabled="true" visible="false" x="379" y="28" width="57" height="22"/>
                  </XCUIElementTypeButton>
                </XCUIElementTypeOther>
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="64" width="375" height="1"/>
              </XCUIElementTypeOther>
              <XCUIElementTypeScrollView type="XCUIElementTypeScrollView" enabled="true" visible="true" x="0" y="0" width="375" height="667">
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
                    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="667"/>
                    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="20" y="287" width="335" height="114">
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="20" y="287" width="335" height="66">
                        <XCUIElementTypeTextView type="XCUIElementTypeTextView" value="I am an alert" enabled="true" visible="true" x="20" y="287" width="335" height="66"/>
                      </XCUIElementTypeOther>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="20" y="353" width="335" height="1"/>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="262" y="353" width="93" height="48">
                        <XCUIElementTypeButton type="XCUIElementTypeButton" name="Close" label="Close" enabled="true" visible="true" x="286" y="360" width="45" height="34"/>
                      </XCUIElementTypeOther>
                    </XCUIElementTypeOther>
                  </XCUIElementTypeOther>
                  <XCUIElementTypeWebView type="XCUIElementTypeWebView" enabled="true" visible="false" x="0" y="0" width="375" height="667">
                    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="64" width="375" height="560"/>
                  </XCUIElementTypeWebView>
                </XCUIElementTypeOther>
              </XCUIElementTypeScrollView>
            </XCUIElementTypeOther>
            <XCUIElementTypeToolbar type="XCUIElementTypeToolbar" enabled="true" visible="true" x="0" y="623" width="375" height="44">
              <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="622" width="375" height="1"/>
              <XCUIElementTypeButton type="XCUIElementTypeButton" name="back" label="Back" enabled="true" visible="true" x="0" y="625" width="43" height="40"/>
              <XCUIElementTypeButton type="XCUIElementTypeButton" name="Forward" label="Forward" enabled="false" visible="true" x="85" y="625" width="42" height="40"/>
              <XCUIElementTypeButton type="XCUIElementTypeButton" name="Share" label="Share" enabled="true" visible="true" x="161" y="624" width="52" height="40"/>
              <XCUIElementTypeButton type="XCUIElementTypeButton" name="Show Bookmarks" label="Show Bookmarks" enabled="true" visible="true" x="239" y="625" width="57" height="40"/>
              <XCUIElementTypeButton type="XCUIElementTypeButton" name="Pages" label="Pages" enabled="true" visible="true" x="318" y="625" width="58" height="40"/>
            </XCUIElementTypeToolbar>
          </XCUIElementTypeOther>
        </XCUIElementTypeOther>
      </XCUIElementTypeOther>
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="374" y="0" width="2" height="667">
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="374" y="20" width="2" height="647"/>
      </XCUIElementTypeOther>
    </XCUIElementTypeOther>
  </XCUIElementTypeWindow>
  <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="false" x="0" y="0" width="375" height="667">
    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="667"/>
  </XCUIElementTypeWindow>
  <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="true" x="0" y="0" width="375" height="667">
    <XCUIElementTypeStatusBar type="XCUIElementTypeStatusBar" enabled="true" visible="true" x="0" y="0" width="375" height="20">
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="20"/>
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="20">
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="6" y="0" width="39" height="20"/>
        <XCUIElementTypeOther type="XCUIElementTypeOther" value="SSID" name="3 of 3 Wi-Fi bars" label="3 of 3 Wi-Fi bars" enabled="true" visible="true" x="50" y="0" width="13" height="20"/>
        <XCUIElementTypeOther type="XCUIElementTypeOther" name="4:41 PM" label="4:41 PM" enabled="true" visible="true" x="165" y="0" width="48" height="20"/>
        <XCUIElementTypeOther type="XCUIElementTypeOther" name="-100% battery power" label="-100% battery power" enabled="true" visible="true" x="345" y="0" width="25" height="20"/>
      </XCUIElementTypeOther>
    </XCUIElementTypeStatusBar>
  </XCUIElementTypeWindow>
</XCUIElementTypeApplication></AppiumAUT>
```

**Comment by mykola-mokhnach** [comment]
Added the condition for iOS 10

**Comment by imurchie** [comment]
Great! Seems to work.

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**